packages/jsii-pacmak/lib/targets/version-utils.ts

```diff
@@ -137,6 +229,6 @@ function toBracketNotation(
     if (trimDashZero) {
       str = str.replace(/-0$/, '');
     }
-    return suffix ? `${str}${suffix}` : str;
+    return suffix ? `${str}${suffix}` : toReleaseVersion(str, target);
```

Comment:
The existing implementation looks as though one form of version string (semver) is being converted to another form (bracket form?). 
Perhaps leave this translation 'pure' and move the call to `toReleaseVersion()` one level above?

Reply:
The bracket form is a *range* transformation... And the range has versions on both sides, which need correct handling.